internal/services/monitor/monitor_autoscale_setting_resource.go

```diff
@@ -441,13 +455,22 @@ func resourceMonitorAutoScaleSettingCreateUpdate(d *pluginsdk.ResourceData, meta
 
 	t := d.Get("tags").(map[string]interface{})
 
+	predictiveAutoscalePolicy := autoscalesettings.PredictiveAutoscalePolicy{
```

Comment:
> For existing resources, does the service also return the predictive policy with scale mode set to Disabled by default?

Yes, and the `predictiveAutoscalePolicy` block with `"scaleMode": "Disabled"` is always returned by API if we don't set it or for existing resources. Since [Computed+Optional](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/best-practices.md#setting-properties-to-optional--computed) or new Required property is not allowed, the block is flattened.

Reply:
I think it should be fine to make it to be an optional block. Though it is a breaking change for users - they'll need to either change the configuration to conform to the remote, or *ignore* it. Since on the other hand, the flattend schema can also introduce breaking changes if users have out-of-band settings to these new properties (e.g. set the scale mode to `Enabled`).

Reply:
@magodo since `predictiveScaleMode` has a default value, we can fix that behaviourally however. If it's set to `Disabled` the block should be omitted during flattening, else we should output it - however the value `Disabled` shouldn't be exposed in Terraform, instead using the omission of this field to mean `Disabled` (during the expand) - which'll get over the (presumable) tri-state that exists for existing resources (note, newly provisioned resources _on this API Version_ will have a default value - the same isn't necessarily true when using an older version)

Reply:
@teowa Can we follow up tom's comment above to update the flatten function and remove the `Disabled` state for the scale mode from schema? Also, it's worth to verify that resources created by older API versions also always return the block even if it is disabled.

Reply:
> it's worth to verifying that resources created by older API versions also always return the block even if it is disabled.

Verified older API version resources also always return the block if it is disabled.